File: frontend/src/types/command.ts
export type CommandType =
  | 'SET_MODE_STANDBY'
  | 'SET_MODE_TRACKING'
  | 'SET_MODE_READY'
  | 'ABORT_SIMULATION'
  | 'RESET_FAULT'
  | 'RUN_SYSTEM_CHECK';

export type CommandStatus =
  | 'SENDING'
  | 'ACK_RECEIVED'
  | 'ACK_TIMEOUT'
  | 'REJECTED_INVALID_TRANSITION'
  | 'EXECUTED';

export interface CommandPayload {
  command_type: CommandType;
  requested_by: string;
  reason: string;
}

export interface CommandAcknowledgement {
  command_id: string;
  command_type: CommandType;
  status: CommandStatus;
  timestamp: string;
  previous_mode?: string;
  new_mode?: string;
  message?: string;
}

File: backend/src/models/command.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};

/// Type of command sent by the operator.
#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
pub enum CommandType {
    SetModeStandby,
    SetModeTracking,
    SetModeReady,
    AbortSimulation,
    ResetFault,
    RunSystemCheck,
}

/// A command request sent to `POST /commands`.
#[derive(Debug, Clone, Deserialize)]
pub struct CommandRequest {
    pub command_type: CommandType,
    pub requested_by: String,
    pub reason: String,
}

/// Status of a submitted command.
#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
pub enum CommandStatus {
    AckReceived,
    AckTimeout,
    RejectedInvalidTransition,
    Executed, // Internal use
}

/// HTTP response for `POST /commands`.
#[derive(Debug, Clone, Serialize)]
pub struct CommandResponse {
    pub command_id: String,
    pub command_type: CommandType,
    pub status: CommandStatus,
    pub previous_mode: String,
    pub new_mode: String,
    pub timestamp: DateTime<Utc>,
    pub message: String,
}

/// A command's tracking state within the simulator.
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CommandRecord {
    pub command_id: String,
    pub command_type: CommandType,
    pub requested_by: String,
    pub status: CommandStatus,
    pub timestamp: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}
